Source/AliveLibAO/Bullet.cpp

```diff
@@ -35,4 +43,202 @@ BaseGameObject* Bullet::VDestructor(signed int flags)
+                ) == 1)
+            {
+                FP distHit;
+                FP distShot;
+                if (pShotObj)
+                {
+                    distHit = hitX - field_18_xpos;
+                    if (hitX - field_18_xpos < FP_FromInteger(0))
+                    {
+                        distHit = field_18_xpos - hitX;
+                    }
+                    distShot = pShotObj->field_A8_xpos - field_18_xpos;
+                    if (distShot < FP_FromInteger(0))
+                    {
+                        distShot = field_18_xpos - pShotObj->field_A8_xpos;
+                    }
+                }
+                if (!pShotObj || (distShot > distHit))
+                {
+                    if (field_20 <= FP_FromInteger(0))
+                    {
+                        auto pSpark = ao_new<Spark>();
+                        if (pSpark)
+                        {
+                            pSpark->ctor_477B70(hitX, hitY, field_2C_scale, 6u, -76, 76);
+                        }
+                    }
+                    else
+                    {
+                        auto pSpark = ao_new<Spark>();
+                        if (pSpark)
+                        {
+                            pSpark->ctor_477B70(hitX, hitY, field_2C_scale, 6u, 50, 205);
+                        }
+                    }
+                    New_Particles_419A80(hitX, hitY, field_2C_scale, 3, 0);
+                    if (Math_RandomRange_450F20(0, 100) >= 90)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    if (Math_RandomRange_450F20(0, 128) < 64)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    SFX_Play_43AD70(SoundEffect::Unknown_2, volume, 0);
+                    field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                    return;
+                }
+            }
+            else
+            {
+                if (!pShotObj)
+                {
+                    if ((signed __int16) Math_RandomRange_450F20(0, 100) >= 70)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    if (Math_RandomRange_450F20(0, 128) < 64)
+                    {
+                        SFX_Play_43AD70(SoundEffect::Unknown_1, volume, 0);
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    SFX_Play_43AD70(SoundEffect::Unknown_2, volume, 0);
+                    field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                    return;
+                }
+            }
+            if (pShotObj->VTakeDamage_401920(this))
```

Comment:
should be  if (pShotObj->VTakeDamage(this)) without the address else it will be a static rather than a virtual call